data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/activities/TemplateParamsUpgradePlugin.java

```diff
@@ -0,0 +1,91 @@
+package org.exoplatform.ecms.upgrade.activities;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.commons.persistence.impl.EntityManagerService;
+import org.exoplatform.commons.upgrade.UpgradePluginExecutionContext;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.container.ExoContainerContext;
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.container.component.RequestLifeCycle;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+public class TemplateParamsUpgradePlugin extends UpgradeProductPlugin {
+  private static final Log LOG = ExoLogger.getExoLogger(TemplateParamsUpgradePlugin.class);
+
+  private final PortalContainer container;
+
+  private final EntityManagerService entityManagerService;
+
+  private int                  templatePramasUpdatedCount;
+
+  private static final String  OLD_TEMPLATE_PARAMS_KEY = "WORKSPACE  ";
+
+  private static final String  NEW_TEMPLATE_PARAMS_KEY = "WORKSPACE";
```

Comment:
Could you pass those params as Initparams of the upgrade plugin, this way this UP may be used to replace values of other template params keys .

Reply:
done !